src/super_gradients/training/transforms/keypoints/keypoints_mosaic.py

```diff
@@ -0,0 +1,132 @@
+            right = self.pad_sample(right, pad_top=max_height - right.image.shape[0])
+        else:
+            left = self.pad_sample(left, pad_bottom=max_height - left.image.shape[0])
+            right = self.pad_sample(right, pad_bottom=max_height - right.image.shape[0])
+
+        image = np.concatenate([left.image, right.image], axis=1)
+        mask = np.concatenate([left.mask, right.mask], axis=1)
+
+        left_sample_width = left.image.shape[1]
+
+        right_bboxes = right.bboxes
+        if right_bboxes is None:
+            right_bboxes = np.zeros((0, 4), dtype=np.float32)
+
+        right_joints_offset = np.array([left_sample_width, 0, 0], dtype=right.joints.dtype).reshape((1, 1, 3))
+        right_bboxes_offset = np.array([left_sample_width, 0, 0, 0], dtype=right_bboxes.dtype).reshape((1, 4))
+
+        joints = np.concatenate([left.joints, right.joints + right_joints_offset], axis=0)
+        bboxes = self._concatenate_arrays(left.bboxes, right_bboxes + right_bboxes_offset, shape_if_empty=(0, 4))
+
+        is_crowd = np.concatenate([left.is_crowd, right.is_crowd], axis=0)
+        areas = self._concatenate_arrays(left.areas, right.areas, shape_if_empty=(0,))
+        return PoseEstimationSample(image=image, mask=mask, joints=joints, is_crowd=is_crowd, bboxes=bboxes, areas=areas, additional_samples=None)
+
+    def stack_samples_vertically(self, top, bottom):
+        max_width = max(top.image.shape[1], bottom.image.shape[1])
+
+        pad_left = (max_width - top.image.shape[1]) // 2
+        pad_right = max_width - top.image.shape[1] - pad_left
+        top = self.pad_sample(top, pad_left=pad_left, pad_right=pad_right)
+
+        pad_left = (max_width - bottom.image.shape[1]) // 2
+        pad_right = max_width - bottom.image.shape[1] - pad_left
+        bottom = self.pad_sample(bottom, pad_left=pad_left, pad_right=pad_right)
+
+        image = np.concatenate([top.image, bottom.image], axis=0)
+        mask = np.concatenate([top.mask, bottom.mask], axis=0)
+
+        top_sample_height = top.image.shape[0]
+
+        bottom_bboxes = bottom.bboxes
+        if bottom_bboxes is None:
+            bottom_bboxes = np.zeros((0, 4), dtype=np.float32)
+
+        bottom_joints_offset = np.array([0, top_sample_height, 0], dtype=bottom.joints.dtype).reshape((1, 1, 3))
+        bottom_bboxes_offset = np.array([0, top_sample_height, 0, 0], dtype=bottom_bboxes.dtype).reshape((1, 4))
+
+        joints = np.concatenate([top.joints, bottom.joints + bottom_joints_offset], axis=0)
+        bboxes = self._concatenate_arrays(top.bboxes, bottom_bboxes + bottom_bboxes_offset, shape_if_empty=(0, 4))
+
+        is_crowd = np.concatenate([top.is_crowd, bottom.is_crowd], axis=0)
+        areas = self._concatenate_arrays(top.areas, bottom.areas, shape_if_empty=(0,))
+        return PoseEstimationSample(image=image, mask=mask, joints=joints, is_crowd=is_crowd, bboxes=bboxes, areas=areas, additional_samples=None)
```

Comment:
So this transform is not deterministic in output size?
I think this should be documented.

Just another note:
I am not gonna be a party pooper, because it is nice work (:
I do think there is value to having deterministic output size while having having the mosaic center being randomly sampled as we do in DetectionMosaic.